### ISSUE TITLE
Enhancement: support upload of data objects

### ISSUE THREAD

**Issue by ncullen93** [JohnCoene/firebase#37]
Currently seems like only files can be uploaded since 'data_url' is passed to the `uploadString` function from firebase SDK. This could be handled by using `uploadBytes` or by setting the file type argument to 'base64' instead of 'data_url'

Reference: https://firebase.google.com/docs/storage/web/upload-files

My current solution is to just write the data to a temp file before upload and then delete it afterwards, but that's not ideal. I may tackle this and submit a pr if it really becomes an issue.

**Comment by JohnCoene** [comment]
Yes, that should be added. Sorry, I'm a bit overwhelmed with work at the moment and have little time for OSS, happy to look at a PR thought.
